mico-admin/src/app/api/api.service.ts

```diff
@@ -986,4 +1012,116 @@ export class ApiService {
             filter(data => data !== undefined)
         );
     }
+
+
+    // =======
+    // POLLING
+    // =======
+
+
+    /**
+     * Polls the deployment job of an given application and provides feedback if the deployment failed/ was successful.
+     * @param applicationShortName applicationShortName of the application to be polled
+     * @param applicationVersion applicationVersion of the application to be polled
+     */
+    pollDeploymentJobStatus(applicationShortName: string, applicationVersion: string) {
+        const resource = 'poll/jobs/' + applicationShortName + '/' + applicationVersion + '/status';
+        const stream = this.getStreamSource<any>(resource);
+
+        // poll status, end polling after 3 minutes
+        const subPolling = interval(500).pipe(takeUntil(timer(3 * 60 * 1000)))
+            .subscribe(() => {
+                this.getJobStatus(applicationShortName, applicationVersion);
+            });
+
+        // handle incomming status updates
+        const subJobStatus = this.getJobStatus(applicationShortName, applicationVersion).subscribe(newStatus => {
+
+            if (newStatus.status === 'DONE') {
+
+                this.snackBar.open('Application deployment finished: ' +
+                    applicationShortName + ' ' + applicationVersion, 'Ok', {
+                        duration: 4000,
+                    });
+
+                // cleanup and return
+                safeUnsubscribe(subPolling);
+                safeUnsubscribe(subJobStatus);
+                // object freeze results in undefined, hence deepcopy
+                stream.next(JSON.parse(JSON.stringify(newStatus)));
+
+            } else if (newStatus.status === 'ERROR') {
+                this.snackBar.open('Application deployment failed: ' +
+                    applicationShortName + ' ' + applicationVersion, 'Ok', {
+                        duration: 8000,
+                    });
+
+                // cleanup and return
+                safeUnsubscribe(subPolling);
+                safeUnsubscribe(subJobStatus);
+                stream.next(JSON.parse(JSON.stringify(newStatus)));
+            }
+
+        });
+
+        return stream.asObservable().pipe(
+            filter(data => data !== undefined)
+        );
+
+    }
+
+    /**
+     * Subscribes ,to the getApplicationDeploymentStatus for a specified application
+     * The polling is started via startApplicationStatusPolling
+     *
+     * @param applicationShortName shortName of the application
+     * @param applicationVersion version of the application
+     */
+    pollApplicationStatus(applicationShortName: string, applicationVersion: string) {
```

Comment:
remove this method and directly use `getApplicationDeploymentStatus` in the components